Sources/OktaDirectAuth/DirectAuthFlow.swift

```diff
@@ -22,8 +22,15 @@ public protocol DirectAuthenticationFlowDelegate: AuthenticationDelegate {
 }
 
 public enum DirectAuthenticationFlowError: Error {
+    case invalidOTP
+    case oobRejected
+    case invalidChallengeTypesSupported
+    case pollingTimeoutExceeded
     case missingArguments(_ names: [String])
-    case currentStatusMissing
+    case network(error: APIClientError)
+    case oauth2(error: OAuth2Error)
+    case server(error: OAuth2ServerError)
+    case other(error: Error)
```

Comment:
Just tested and I got `.other(error: )` in invalid OTP scenario, and the associated value inside is also a `DirectAuthenticationFlowError. other(error:)`
![Screen Shot 2023-03-14 at 2 07 49 PM](https://user-images.githubusercontent.com/77817162/225137100-36a12fe8-560e-4d74-bd0a-013232e47916.png)

Reply:
Hmm, so that's not working as expected.  OAuth2ServerError should be returned as a FlowError `.server(error:)`. 

Also I think there are server limitations which are causing the wrong error code to be returned. So until those APIs get updated, the client SDK code might require some work-arounds.